Sources/ClientRuntime/EventStream/MessageUnmarshallable.swift

```diff
@@ -16,3 +16,14 @@ public protocol MessageUnmarshallable {
     ///              using the same decoder.
     init(message: EventStream.Message, decoder: ResponseDecoder) throws
 }
+
+public typealias UnmarshalClosure<T> = (EventStream.Message) throws -> T
```

Comment:
Defines the closure type for decoding an event in a stream, and provides an implementation for JSON that uses a Swift decoder.